### ISSUE TITLE
Linker error when compiling for ARM64

### ISSUE THREAD

**Issue by jdpatdiscord** [processhacker/processhacker#1110]
### Brief description of your issue

![image](https://user-images.githubusercontent.com/42700985/150954424-743e9e91-29ba-419a-a321-05f3ee89507e.png)

This error appears when building for ARM64. It does not appear when building on the Win32/x64 profiles.
I am working on porting Process Hacker to ARM64, modifying multiple source files to do so, and this is one I cannot get past.

### Steps to reproduce (optional)

1. Clone Process Hacker
2.  Open the .sln in Visual Studio 2022 (you will run into errors not relevant on vs2019)
3. Select the 'ARM64' build profile
4. Rebuild `tools\thirdparty\thirdparty.sln` for ARM64 to fix a different error
5. Spam `#if defined(_M_X64) || defined(_M_IX86)` /  `#elif defined(_M_ARM64)` where architecture-specific code is (in order to reach this specific build error quickly, and if you don't know where the compiler will tell you with errors)

### Expected behavior (optional)

For it to compile or present me with errors other than this one to troubleshoot

**Comment by dmex** [comment]
>   Creating library \bin\DebugARM64\ProcessHacker.lib and object \bin\DebugARM64\ProcessHacker.exp
>ProcessHacker.vcxproj -> \bin\DebugARM64\ProcessHacker.exe
> ========== Rebuild All: 2 succeeded, 0 failed, 0 skipped ==========

ARM64 build works on my machine.

> _InterlockedExchangeAddPointer

_InterlockedExchangeAddPointer doesn't exist in those LIBCMT lib/obj files.

> Rebuild tools\thirdparty\thirdparty.sln for ARM64 to fix a different error

The \build\build_tools.cmd script compiles the thirdparty libs and the ARM64 version compiled without errors, so I don't know what error you had to fix but it's probably broken the library.

1) Run `\build\build_tools.cmd`
2) Run `\build\build_arm64.cmd`

This will compile everything for ARM64... @MagicAndre1981 or @Biswa96 might be able to help.

**Comment by MagicAndre1981** [comment]
I don't have the C++ workload installed in my 17.1 preview VM, so I can't test it.

**Comment by jdpatdiscord** [comment]
![image](https://user-images.githubusercontent.com/42700985/151370144-26725cc4-1a79-4104-8761-9ad9ca152942.png)

What's this? I just installed Vs2022.

**Comment by Biswa96** [comment]
Do you have Windows Driver Kit installed?

**Comment by jdpatdiscord** [comment]
> Do you have Windows Driver Kit installed?

Yes, I installed it from https://docs.microsoft.com/en-us/windows-hardware/drivers/download-the-wdk

Currently Microsoft doesn't provide WDK installer support for Visual Studio 2022. What should I do?

**Comment by InternetPseudonym** [comment]
> Currently it appears Microsoft doesn't provide WDK installer support for Visual Studio 2022. What should I do?

use Visual Studio 2019. 2022 is a bit unfinished, it seems

**Comment by jdpatdiscord** [comment]
![image](https://user-images.githubusercontent.com/42700985/151552937-baa4a871-2af3-4f6f-a812-9cfe24e80e83.png)

I managed to get it to build using VS2019 build tools after modifying the build scripts to do so, and then had to disable spectre mitigated libs across all projects to get rid of the 'spectre mitigated libs not present' error despite me having installed Arm64, x86 and x64 libs for it in both VS2019 and VS2022 via the installer, AND the build tools variants.

These errors were present when building, but I will make a pull request to fix them after testing the changes.
Long story short, the ARM64 build process quickly breaks down if anyone else is building them, and Microsoft themselves is lagging behind on WDK integration for VS2022.

**Comment by jdpatdiscord** [comment]
Okay. With the original problem being solved, I'll close the issue. There is a new pull request to fix the compiler errors: [https://github.com/processhacker/processhacker/pull/1113](url)
